TypeProvider/Core/Naming.fs

```diff
@@ -0,0 +1,30 @@
+﻿[<RequireQualifiedAccess>]
+module internal Froto.TypeProvider.Core.Naming
```

Comment:
I just meant moving [`TypeResolver.resolveScalar`](https://github.com/ctaggart/froto/pull/44/files#diff-ee5d67339d39c24949db651a42bff2b8R45) function to more central place so that it can be re-used in other parts of Froto